app/models/miq_cockpit_ws_worker.rb

```diff
@@ -0,0 +1,118 @@
+
+  self.required_roles = ['cockpit_ws']
+  self.maximum_workers_count = 1
+
+  def friendly_name
+    @friendly_name ||= "Cockpit Worker"
+  end
+
+  def self.can_activate_cockpit? (opts = nil)
+    if self.has_required_role?
+      if (!opts)
+          opts = MiqCockpitWsWorker.worker_settings()
+      end
+
+      if opts[:external_url]
+        return true
+      else
+        return @supports_cockpit_ws unless @supports_cockpit_ws.nil?
+        @supports_cockpit_ws = MiqEnvironment::Command.is_linux? && File.exist?('/usr/libexec/cockpit-ws') && File.exist?('/usr/libexec/cockpit-ssh')
+        return @supports_cockpit_ws
+      end
+    else
+      return false
+    end
+  end
+
+
+  # Update settings for other servers
+  def self.update_other_server_configs(opts)
+    url = opts[:external_url]
+    if (!url)
+      url = MiqCockpit::WS.url(opts, MiqServer.my_server.ipaddress)
+    end
+
+    MiqServer.active_miq_servers().where.not(:id => MiqServer.my_server.id) do |server|
+      conf = server.get_config()
+      conf[:workers][:cockpit_ws_worker] = {
+        :external_url => url
+      }
+      server.set_config(conf)
+    end
+  end
+
+  def self.sync_workers
+    opts = MiqCockpitWsWorker.worker_settings()
+    install_apache_proxy_config(opts) if MiqEnvironment::Command.supports_apache?
+    @workers = 0 unless self.can_activate_cockpit?(opts)
+    @workers = 1 unless opts[:external_url]
+    ret = super
+
+    if self.can_activate_cockpit?(opts)
+      self.update_other_server_configs (opts)
+    end
+    return ret
+  end
+
+  def self.install_apache_proxy_config(opts = nil)
+    if (!opts)
+        opts = MiqCockpitWsWorker.worker_settings()
+    end
+    if self.has_required_role? || opts[:external_url]
+      MiqCockpit::ApacheConfig.new(opts).save(APACHE_CONF_FILE)
+    elsif File.exists?(APACHE_CONF_FILE)
+      File.truncate(APACHE_CONF_FILE, 0)
+    end
+    MiqServer.my_server.queue_restart_apache if MiqEnvironment::Command.supports_apache?
+  end
+
+  def terminate
+    begin
+      _log.info("Terminating #{format_full_log_msg}, status [#{status}]")
+      Process.kill("TERM", pid)
+      # TODO: Variablize and clean up this 10-second-max loop of waiting on Worker to gracefully shut down
+      10.times do
+        unless MiqProcess.alive?(pid)
+          update_attributes(:stopped_on => Time.now.utc, :status => MiqWorker::STATUS_STOPPED)
+          break
+        end
+        sleep 1
+      end
+    rescue Errno::ESRCH
+      _log.warn("#{format_full_log_msg} has been killed")
+    rescue => err
+      _log.warn("#{format_full_log_msg} has been killed, but with the following error: #{err}")
+    end
+
+    kill if MiqProcess.alive?(pid)
+  end
+
+  def start
+    opts = MiqCockpitWsWorker.worker_settings()
+    cockpit_ws = MiqCockpit::WS.new(opts)
+    cockpit_ws.save_config()
+    env = {"XDG_CONFIG_DIRS"=> cockpit_ws.config_dir}
+    pid = Kernel.spawn(env, *cockpit_ws.command(BINDING_ADDRESS))
+    Process.detach(pid)
```

Comment:
The other workers I've seen in digging around the code all are ruby processes so forking there makes sense. Since this is a standalone process that shouldn't inherit anything from the parent, i figured a simple spawn was easier, rather than having a special runner just to do that spawn. But maybe I'm missing something (heartbeats?) so if you prefer that I spawn this from a runner I can.